docs/proposals/kubelet-systemd.md

```diff
@@ -0,0 +1,277 @@
+
+If the `kubelet` is parented by a `slice` that is categorized as system-reserved,
+the `kubelet` will log a warning that kubernetes-reserved observed usage accounting is
+disabled, but the static reservation is still observed for purposes of reporting the
+allocable resources for the node.
+
+If the `kubelet` is launched directly from a terminal, it's most likely destination will
+be in a `scope` that is a child of `user.slice` as follows:
+
+`/sys/fs/cgroup/<controller>/user.slice/user-1000.slice/session-1.scope`
+
+In this context, the parent `scope` is what will be used to facilitate local developer
+debugging scenarios for tracking `kube-reserved` usage.
+
+### Kubernetes container runtime reserved
+
+This proposal asserts that the reservation of compute resources for any associated
+container runtime daemons is tracked by the operator under the system-reserved or
+kubernetes-reserved values at this time and any enforced limits are set by the
+operator specific to the container runtime.
+
+For future consideration, if the `kubelet` attempts to enforce `--kube-reserved`, this
+proposal asserts that the reservation associated with any particular container runtime
+should be split out at that time.  In a `systemd` environment, this proposal asserts that
+you cannot mandate the container runtime daemon is in the same `slice` as the Kubernetes
+components.
+
+**Docker**
+
+If the `kubelet` is configured with the `container-runtime` set to `docker`, the
+`kubelet` will detect the `service` associated with the `docker` daemon and use that
+to do local node accounting.  If an operator wants to impose runtime limits on the
+`docker` daemon to control resource usage, the operator should set those explicitly in
+the `service` unit that launches `docker`.  The `kubelet` will not set any limits itself
+at this time and will assume whatever budget was set aside for `docker` was included in
+either `--kube-reserved` or `--system-reserved` reservations.
+
+**Rocket**
+
+Rocket has no client/server daemon, and therefore has no explicit requirements on container-runtime
+reservation.
+
+### Node allocable
+
+The proposal makes no changes to the definition.
+
+The node will report a set of allocable compute resources defined as follows:
+
+`[Allocatable] = [Node Capacity] - [Kube-Reserved] - [System-Reserved]`
+
+### OOM Score Adjustment
+
+The `kubelet` at bootstrapping will set the `oom_score_adj` value for Kubernetes
+daemons, and any dependent container-runtime daemons.
+
+If `container-runtime` is set to `docker`, then set its `oom_score_adj=-900`
+
+### Linux Kernel Parameters
+
+The `kubelet` will set the following:
+
+* `sysctl -w vm.overcommit_memory=1`
+* `sysctl -w vm.panic_on_oom=0`
+* `sysctl -w vm.overcommit_memory=1`
+* `sysctl -w kernel/panic=1`
+* `sysctl -w kernel/panic_on_oops=10`
+
+## kubelet cgroup-root behavior under systemd
+
+The `kubelet` supports a `cgroup-root` flag which is the optional root `cgroup` to use for pods.
```

Comment:
I thought it appropriate to document how the flag would behave since its related.  I was not sure I wanted to actually advocate it being deprecated at this time in this PR.